packages/api/test/e2e/promise-queries.spec.ts

```diff
@@ -177,20 +177,197 @@ describe.skip('Promise e2e queries', () => {
         done();
-      })
-    );
+      });
+    });
+
+    it('queries correct hash', async () => {
+      const hash = await api.query.balances.freeBalance.hash(keyring.alice.address());
+
+      expect(hash).toBeDefined();
+    });
+
+    it('gets correct key', async () => {
+      // assume the account Alice is only used in test(the balance of Alice does not change in this test case)
+      const key = api.query.balances.freeBalance.key(keyring.alice.address());
+      const balanceData = await api.rpc.state.getStorage(key) as Option<any>;
+      const balanceRPC = new Balance(balanceData.unwrapOr(undefined));
+
+      const balance = await api.query.balances.freeBalance(keyring.alice.address());
+
+      expect(balanceRPC.isZero()).toBe(false);
+      expect(balanceRPC.toString()).toEqual(balance.toString());
+    });
+
+    it('queries multiple results', async () => {
+      // assume the account Alice and Bob are only used in test(the balance of them do not change in this test case)
+      const balanceAlice = await api.query.balances.freeBalance(keyring.alice.address());
+      const balanceBob = await api.query.balances.freeBalance(keyring.bob.address());
+
+      const balances = await api.query.balances.freeBalance.multi([
+        keyring.alice.address(),
+        keyring.bob.address()
+      ]);
+
+      expect(balances).toHaveLength(2);
+      expect((balances as any)[0].toString()).toEqual(balanceAlice.toString());
+      expect((balances as any)[1].toString()).toEqual(balanceBob.toString());
+    });
+
+    it('subscribes to multiple queries and get correct results', async (done) => {
+      // assume the account Alice and Bob are only used in test(the balance of them do not change in this test case)
+      const balanceAlice = await api.query.balances.freeBalance(keyring.alice.address());
+      const balanceBob = await api.query.balances.freeBalance(keyring.bob.address());
+
+      return api.query.balances.freeBalance.multi([
+        keyring.alice.address(),
+        keyring.bob.address()
+      ], (balances) => {
+        expect(balances).toHaveLength(2);
+        expect(balances[0].toString()).toEqual(balanceAlice.toString());
+        expect(balances[1].toString()).toEqual(balanceBob.toString());
+        done();
+      });
+    });
+
+    it('queries correct size', async () => {
+      const size = await api.query.balances.freeBalance.size(keyring.alice.address());
+
+      expect(size.toNumber()).not.toEqual(0);
+    });
+  });
+
+  // TODO Update ['any', '0x1234'] to the key of a known event topic and update EXPECTED_VALUE to the expected value
+  describe('with double map type', async () => {
+    const KEY1 = 'any';
+    const KEY2 = '0x1234';
+    it('queries correct value', async () => {
+      const eventTopics = await api.query.system.eventTopics(KEY1, KEY2);
+
+      expect(eventTopics.toJSON()).toEqual([]);
+    });
+
+    it('queries correct value at a specified block', async () => {
```

Comment:
Here is a special case: `await api.query.system.eventTopics.at(header.hash, KEY1, KEY2);` will throw an `Encoding for input doesn't match output, created 0x00 from 0x` error. This error is from [here](https://github.com/polkadot-js/api/blob/761682373def42dd64587832c1ebd554340ed1a8/packages/types/src/codec/createType.ts#L244). This `at` function only throw this error when the returned value from the chain is `null`. If the returned value is not null, this function will work. I added a unit test below for this issue